### PR TITLE
Containerd-Cri with supports multi-architectures

### PR DESCRIPTION
**Pull request by jr-santos98**
While the containerd build a file getting information about the architecture and the operating system it is running on. The cri assumes that you are using a Linux computer with x86. This commit makes the CRI adapt to the system and architecture of the computer on which it is being used.

### CODE REVIEW

**Comment by mikebrow** [comment]
please squash and sign your commits

**Comment by jr-santos98** [comment]
@mikebrow How do I do this? Failure in the file would be for commit validation.

**Comment by mikebrow** [comment]
normally when modifying a prior commit to make more changes.. I would:
git add filename
git commit -s --amend

To squash if you didn't amend the second commit you can:
git rebase -i HEAD~2
you want the top commit to say pick and just change the second one to squash...
save and exit
git push -f branchname

Cheers, Mike

**Comment by mikebrow** [comment]
to configure your git for signing with the -s option:
$ git config --global user.name "full name here"
$ git config --global user.email "email address here"

**Comment by AkihiroSuda** [comment]
ping @jr-santos98

**Comment by jr-santos98** [comment]
@mikebrow What's the problem now ?? Me running the command locally. It is informed that everything is ok.
The [link](http://www.students.ic.unicamp.br/~ra195727/log.txt) shows the output for the `make .gitvalidation` command that I obtained locally.

**Comment by mikebrow** [comment]
> @mikebrow What's the problem now ?? Me running the command locally. It is informed that everything is ok.
> The [link](http://www.students.ic.unicamp.br/~ra195727/log.txt) shows the output for the `make .gitvalidation` command that I obtained locally.

nothing.. the github action host encountered a problem pulling the files in the pull request threw a 128 network error  then github "checks" code got in a bad state .. 

so I closed/reopened the PR to force the state reset (as github refused to restart the failed action due to an internal error)

sigh.

File: hack/release.sh [hunk 1]
```diff
@@ -42,6 +42,8 @@ OFFICIAL_RELEASE=${OFFICIAL_RELEASE:-false}
 # LOCAL_RELEASE indicates that containerd has been built and released
 # locally.
 LOCAL_RELEASE=${LOCAL_RELEASE:-false}
+GOOS=$(go env GOOS)
```
Comment (mikebrow) [inline hack/release.sh:45]: We may get these from elsewhere so let's only set these if not already set..
for example: `GOOS=$(GOOS:-go env GOOS)`

Comment (jr-santos98) [reply hack/release.sh:45]: I think it's a good idea

Comment (mikebrow) [reply hack/release.sh:45]: bash is so much fun sometimes
```
if [ -z "${GOOS}" ]
then
    GOOS=$(go env GOOS)
fi
```

**Comment by jr-santos98** [comment]
@mikebrow Interesting idea, what do you think of the change I made?

**Comment by jr-santos98** [comment]
@mikebrow I was wrong ... But, I tested it, the condition `if [ -z "${GOOS:-}" ] then  GOOS=$(go env GOOS) fi`.
It will now work and pass the tests.

**Comment by mikebrow** [comment]
```
cat run.sh

#!/bin/bash
if [ -z "${GOOS}" ]
then
    GOOS=$(go env GOOS)
fi
echo $GOOS

$ GOOS=foo ./run.sh
foo

$ ./run.sh
linux
```
what failed?

**Comment by mikebrow** [comment]
the kubernetes e2e stuff is just flakey ATM ... nothing to do with your script..

**Comment by jr-santos98** [comment]
The error "unbound variable" is displayed in some of the CI points.
As can be seen in this [code](https://repl.it/@Yukine_/condition#main.sh).
This error happens while running my build as well. Even without that dependency.
In this format:"${GOOS:-}" ... or: "${GOOS}", there is no error in the tests, and I was able to run locally as well.

**Comment by mikebrow** [comment]
> The error "unbound variable" is displayed in some of the CI points.
> As can be seen in this [code](https://repl.it/@Yukine_/condition#main.sh).
> This error happens while running my build as well. Even without that dependency.
> In this format:"${GOOS:-}" ... or: "${GOOS}", there is no error in the tests, and I was able to run locally as well.

Interesting.. works fine on ubuntu :-) Thx!

**Pull request merged**